### PR TITLE
Various improvements and added ability to read "readonly" and "required" fields flags

### PR DESCRIPTION
**Pull request by Emulator000**
In this PR I ported all the code from @malte-v and introduced some improvements such libraries updating, Clippy and field flags reading.

### CODE REVIEW

**Comment by jsandler18** [comment]
I had forgotten about this little library! If a fork added these improvements, why bother adding them here?

**Comment by Emulator000** [comment]
@jsandler18 because you own the original "pdf_form" library on crates.io and maybe you want to port all these improvements onto your original library.

**Comment by jsandler18** [comment]
¯\_(ツ)_/¯ sure why not.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+**/*.rs.bk
+/*.data
+/*.gz
+/target/
```
Comment (jsandler18) [inline .gitignore:4]: add .idea so all the intellij files are not included?

File: Cargo.toml [hunk 2]
```diff
@@ -8,6 +8,6 @@ description = "A library for programatically filling out pdf forms"
 license = "MIT"
```
Comment (jsandler18) [inline Cargo.toml:9]: I suppose we should be at version 0.3.0, right?

Also, you should probably add yourself and this malte-v to the author list.

File: .gitignore (hunk 1 above)
Comment (Emulator000) [reply .gitignore:4]: Not all IntelliJ IDEA files are usefulness, some contains useful information but I can remove all `.idea` folder if you prefer :-)

File: Cargo.toml (hunk 2 above)
Comment (Emulator000) [reply Cargo.toml:9]: Yes, I'll do it now :-)

**Pull request merged**